beacon-chain/p2p/pubsub_filter.go

```diff
@@ -0,0 +1,81 @@
+package p2p
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	pubsubpb "github.com/libp2p/go-libp2p-pubsub/pb"
+	"github.com/prysmaticlabs/prysm/beacon-chain/p2p/encoder"
+)
+
+var _ pubsub.SubscriptionFilter = (*Service)(nil)
+
+const pubsubSubscriptionRequestLimit = 100
+
+// CanSubscribe returns true if the topic is of interest and we could subscribe to it.
+func (s *Service) CanSubscribe(topic string) bool {
+	if !s.isInitialized() {
+		return false
+	}
+	parts := strings.Split(topic, "/")
+	if len(parts) != 5 {
+		return false
+	}
+	// The topic must start with a slash, which means the first part will be empty.
+	if parts[0] != "" {
+		return false
+	}
+	if parts[1] != "eth2" {
+		return false
+	}
+	fd, err := s.forkDigest()
+	if err != nil {
+		log.WithError(err).Error("Could not determine fork digest")
+		return false
+	}
+	if parts[2] != fmt.Sprintf("%x", fd) {
+		return false
+	}
```

Comment:
maybe a small comment on why we skip `parts[3]`

Reply:
Done